crypto/crypto.go

```diff
@@ -234,15 +238,20 @@ func GeneratePrivateKey(sigAlgo SignatureAlgorithm, seed []byte) (PrivateKey, er
 		seedLen = crypto.KeyGenSeedMinLenECDSAP256
 	case ECDSA_secp256k1:
 		seedLen = crypto.KeyGenSeedMinLenECDSASecp256k1
-        default:
-                 return PrivateKey{}, fmt.Errorf("crypto: invalid signature algorithm %s", sigAlgo)
+	default:
+		return PrivateKey{}, fmt.Errorf(
+			"crypto: key generation not supported for non-ECDSA algorithm %s",
```

Comment:
I think a more precise error message would be that the signature algorithm is just not supported by the SDK. Why: the SDK only supports ECDSA with 2 curves, so we could have other curves implemented with ECDSA but they are still not supported by the SDK.

Reply:
Great point, I'll add 👍